src/model/model.ts

```diff
@@ -220,7 +220,19 @@ export class Model {
       if (!ast) {
         return false;
       }
-      ast.policy.push(rule);
+
+      const policy = ast.policy;
+      const tokens = ast.tokens;
+
+      if (tokens.find((str) => str === 'p_priority')) {
+        const targetPriorityIndex = tokens.indexOf('p_priority');
+        const rulePriority = rule[targetPriorityIndex];
+        const targetInsertIndex = policy.findIndex((oneRule) => oneRule[targetPriorityIndex] >= rulePriority);
+        // If targetInsertIndex returns -1, next line will still work as expected.
```

Comment:
one thing to note that when the targetInsertIndex is -1, the rule will inserted to `policy[policy.length - 1 - 1]`.